franka_state_controller:
  type: franka_control/FrankaStateController
  publish_rate: 50  # [Hz]
  arm_id: $(arg arm_id)
  joint_names:
    - $(arg arm_id)_joint1
    - $(arg arm_id)_joint2
    - $(arg arm_id)_joint3
    - $(arg arm_id)_joint4
    - $(arg arm_id)_joint5
    - $(arg arm_id)_joint6
    - $(arg arm_id)_joint7

cartesian_twist_controller:
    type: safe_placement/CartesianTwistController
    publish_rate: 50  # [Hz]
    arm_id: $(arg arm_id)
    joint_names:
        - $(arg arm_id)_joint1
        - $(arg arm_id)_joint2
        - $(arg arm_id)_joint3
        - $(arg arm_id)_joint4
        - $(arg arm_id)_joint5
        - $(arg arm_id)_joint6
        - $(arg arm_id)_joint7
    gains:
      $(arg arm_id)_joint1: { p: 10000, d: 0, i: 1000 }
      $(arg arm_id)_joint2: { p: 10000, d: 0, i: 1000 }
      $(arg arm_id)_joint3: { p: 10000, d: 0, i: 1000 }
      $(arg arm_id)_joint4: { p: 10000, d: 0, i: 1000 }
      $(arg arm_id)_joint5: { p: 10000, d: 0, i: 1000 }
      $(arg arm_id)_joint6: { p: 10000, d: 0, i: 1000 }
      $(arg arm_id)_joint7: { p: 10000, d: 0, i: 1000 }

cartesian_impedance_example_controller:
  type: franka_example_controllers/CartesianImpedanceExampleController
  publish_rate: 50
  arm_id: $(arg arm_id)
  joint_names:
    - $(arg arm_id)_joint1
    - $(arg arm_id)_joint2
    - $(arg arm_id)_joint3
    - $(arg arm_id)_joint4
    - $(arg arm_id)_joint5
    - $(arg arm_id)_joint6
    - $(arg arm_id)_joint7

cartesian_impedance_controller:
  type: safe_placement/CartesianVelocityImpedanceController
  publish_rate: 50
  arm_id: $(arg arm_id)
  joint_names:
    - $(arg arm_id)_joint1
    - $(arg arm_id)_joint2
    - $(arg arm_id)_joint3
    - $(arg arm_id)_joint4
    - $(arg arm_id)_joint5
    - $(arg arm_id)_joint6
    - $(arg arm_id)_joint7

joint_velocity_example_controller:
    type: franka_example_controllers/JointVelocityExampleController
    arm_id: $(arg arm_id)
    joint_names:
        - $(arg arm_id)_joint1
        - $(arg arm_id)_joint2
        - $(arg arm_id)_joint3
        - $(arg arm_id)_joint4
        - $(arg arm_id)_joint5
        - $(arg arm_id)_joint6
        - $(arg arm_id)_joint7